src/data-list/data-list.service.ts

```diff
@@ -0,0 +1,276 @@
+            var changedCount = 0;
+            for (var param in filter) {
+                if (filter.hasOwnProperty(param)) {
+                    var val = this.$location.search()[param];
+                    if (val) {
+                        try {
+                            filter[param] = JSON.parse(val);
+                            if (DataList.isNumber(filter[param])) {
+                                filter[param] = JSON.stringify(filter[param]);
+                            }
+                        } catch (e) {
+                            var isoDatePattern = /(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+([+-][0-2]\d:[0-5]\d|Z))|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d([+-][0-2]\d:[0-5]\d|Z))|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d([+-][0-2]\d:[0-5]\d|Z))/;
+                            if (isoDatePattern.test(val)) {
+                                filter[param] = new Date(val);
+                            } else {
+                                filter[param] = val;
+                            }
+                        }
+                        changedCount++;
+                    }
+                }
+            }
+            return changedCount;
+        }
+
+        private setLocationParams(filter) {
+            var params = this.getFilterAsUrlParams(filter, true);
+            for (var param in params) {
+                if (params.hasOwnProperty(param)) {
+                    var val = params[param];
+                    if (val && param !== 'expand' && param !== 'skip' && param !== 'limit') {
+                        this.$location.search(param, val);
+                    } else {
+                        this.$location.search(param, null);
+                    }
+                }
+            }
+        }
+
+        private getFilterAsUrlParams(filter, addNulls) {
+            var params = {};
+
+            for (var param in filter) {
+                if (filter.hasOwnProperty(param)) {
+                    var val = filter[param];
+
+                    if (val instanceof Array) {
+                        val = val.join(',');
+                    }
+                    if (val && (DataList.isNumber(val) || val.length > 0)) {
+                        params[param] = val;
+                    } else if ((typeof val) === 'boolean') {
+                        params[param] = val === true;
+                    } else if (val instanceof Date) {
+                        params[param] = val.toISOString();
+                    } else if (val && (typeof val) === 'object') {
+                        params[param] = JSON.stringify(val);
+                    } else if (addNulls) {
+                        params[param] = null;
+                    }
+                }
+            }
+            return params;
+        }
+
+        private static isNumber(n) {
+            return !isNaN(parseFloat(n)) && isFinite(n);
+        }
+    }
+
+    export class DataListService {
```

Comment:
Previously it was factory, but now it have two methods init and get so it is more like service :)